src/main/webapp/script.js

```diff
@@ -1,3 +1,96 @@
+class ParameterInput extends HTMLElement {
```

Comment:
Could you add comments to your functions to explain what they each do? I feel that would make this code a lot more readable.

Reply:
Yeah, that gets added in the next PR.

Reply:
It's better to have Javadocs for methods while creating them.